cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+        if (aclBinding.entry().operation().toString().equals("WRITE")
+            || aclBinding.entry().operation().toString().equals("READ")) {
+          acls.add(aclbindingMap);
+        }
+      }
+    }
+  }
+
+  public synchronized String updateProducerAcl(ClusterAclRequest clusterAclRequest) {
+    log.info("updateProducerAclRequest {}", clusterAclRequest);
+    AdminClient client;
+    try {
+      PatternType patternType;
+      if (clusterAclRequest.isPrefixAcl()) {
+        patternType = PatternType.PREFIXED;
+      } else {
+        patternType = PatternType.LITERAL;
+      }
+
+      client =
+          clusterApiUtils.getAdminClient(
+              clusterAclRequest.getEnv(),
+              clusterAclRequest.getProtocol(),
+              clusterAclRequest.getClusterName());
+      if (client == null) {
+        return ApiResultStatus.FAILURE.value;
+      }
+
+      String host,
+          principal,
+          aclSsl = clusterAclRequest.getAclSsl(),
+          aclIp = clusterAclRequest.getAclIp();
+      if (clusterAclRequest.getAclSsl() != null
+          && clusterAclRequest.getAclSsl().trim().length() > 0) {
+        aclSsl = aclSsl.trim();
+        if (AclIPPrincipleType.PRINCIPAL.name().equals(clusterAclRequest.getAclIpPrincipleType())) {
+          host = "*";
+          principal = "User:" + aclSsl;
+
+          if (RequestOperationType.CREATE.equals(clusterAclRequest.getRequestOperationType())) {
+            if (updateTopicProducerWriteAcls(
+                clusterAclRequest.getTopicName(), client, patternType, host, principal)) {
+              return "Acl already exists. success";
+            }
+          } else {
+            processOtherRequests(clusterAclRequest, client, patternType, host, principal);
+          }
+          updateTransactionalIdAclsForProducer(
+              clusterAclRequest.getTransactionalId(),
+              client,
+              patternType,
+              host,
+              principal,
+              clusterAclRequest.getRequestOperationType().value);
+        }
+      }
+
+      if (aclIp != null && aclIp.trim().length() > 0) {
+        aclIp = aclIp.trim();
+        host = aclIp;
+        principal = "User:*";
+
+        if (clusterAclRequest.getRequestOperationType().equals(RequestOperationType.CREATE)) {
+          if (updateTopicProducerWriteAcls(
+              clusterAclRequest.getTopicName(), client, patternType, host, principal)) {
+            return "Acl already exists. success";
+          }
+        } else {
+          processOtherRequests(clusterAclRequest, client, patternType, host, principal);
+        }
+        // Update transactional id acls
+        updateTransactionalIdAclsForProducer(
+            clusterAclRequest.getTransactionalId(),
+            client,
+            patternType,
+            host,
+            principal,
+            clusterAclRequest.getRequestOperationType().value);
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
it's better to print it in log rather than in system out otherwise it's impossible to find it in future
```suggestion
log.error("Exception: ", e);
```